pkg/minishift/addon/manager/addon_manager.go

```diff
@@ -59,21 +61,28 @@ func NewAddOnManager(baseDir string, configMap map[string]*addon.AddOnConfig) (*
 			continue
 		}
 
-		addOn, err := p.Parse(fullPath)
+		addOns, err := p.Parse(fullPath)
 		if err != nil {
 			_, ok := err.(parser.ParseError)
 			if ok {
-				glog.Warning(fmt.Sprintf("Skipping addon '%s' in '%s' due to parse error: %s", f.Name(), fullPath, err.Error()))
+				fmt.Println(fmt.Sprintf("Skipping addon '%s' in '%s' due to parse error: %s", f.Name(), fullPath, err.Error()))
 				continue
 			} else {
 				return nil, errors.Wrapf(err, "Unable to create addon manager for %s. ", baseDir)
 			}
 		}
+		addOn := addOns[0]
+		if len(addOns) > 1 {
+			if addOns[1] != nil {
+				removeAddon := addOns[1]
+				detectedRemoveAddons[removeAddon.MetaData().Name()] = removeAddon
+			}
+		}
 		setStateAndPriority(addOn, configMap)
 		detectedAddOns[addOn.MetaData().Name()] = addOn
 	}
 
-	return &AddOnManager{baseDir: baseDir, addOns: detectedAddOns}, nil
+	return &AddOnManager{baseDir: baseDir, addOns: detectedAddOns, removeAddon: detectedRemoveAddons}, nil
```

Comment:
IMO this is unfortunate. You basically don't tie the remove commands to the actual addon. Instead you say here is a set of remove and apply add-ons with no tie between them. From a model perspective I would expect this to be part of the AddOn interface. There is for example the Commands function - https://github.com/minishift/minishift/blob/4d6d0d30e8f37982e32c1b2b849ad5d17954e917/pkg/minishift/addon/addon.go#L26-L26. There could for example be a RemoveCommands function. I think from a design pov this is better. In this case one would handle the two separate set of commands in AddOnParser.Parse. This method gets the base directory for the add-on. It is a question of checking whether there is a .addon.remove and parsing its commands.

BTW, I am using '.addon.remove' since I think this is what we should use sticking to the terminology we want to use. These are the commands we want to run when we call `minishift addon remove`. Uninstall is about removing the actual files from the add-on dir.